src/components/Button.tsx

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable prettier/prettier */
+import React, { ButtonHTMLAttributes } from 'react';
+import styled from 'styled-components';
+import { darken } from 'polished';
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  borderColor?: string;
+  backgroundColor?: string;
+  textColor?: string;
+  hoverBackgroundColor?: string;
+  hoverTextColor?: string;
+}
+const Button: React.FC<ButtonProps> = ({
```

Comment:
note que esse botão é muito mais genérico (e complexo) do que é necessário aqui nesse PR. Você está prevendo esse uso num próximo PR? Caso contrário, simplifique e adicionamos complexidade depois.

Reply:
Estou sim, na vdd, eu deixei esse botão mais complexo pq eu sei que ele vai necessitar de tudo isso pra as próximas telas, literalmente esse é o template do único botão existente em toda a aplicação, aí depois de uma análise calma e pensativa, decidi fazer esse botão agora, pois de qualquer forma estaria apenas adiando o inevitável, mas se o senhor quiser eu simplifico ele e guardo o código, só acho que deixar pra fazer ele ficar basicamente igual depois seria só um adiamento desnecessário

Reply:
Entendo.

Estou pensando aqui e queria mesmo ver esse código evoluir de algo simples e depois ir adicionando complexidade e refatorando. 

O mínimo que você precisa é um `StyledButton`, que tem `hoverTextColor`, `hoverBackgroundColor`, `borderColor`, `backgroundColor`.

Poderia renomear esse componente e remover os parâmetros que ainda não tem uso?

Reply:
```suggestion
function Button({...} : StyledButtonProps) {
```

Não funcionaria? https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/

Reply:
No final daria no mesmo sim, devo aplicar ?

Reply:
Eu sinceramente prefiro dessa forma por causa da declaração de tipo, e também pra o eslint não dar o problema de [default props](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md) mas se o senhor preferir assim, tudo bem

Reply:
Mas eu ainda acho melhor dessa forma, pois da outra eu estaria sujeito a não conseguir usar propriedades não obrigatórias

Reply:
eu teria que estar definindo default props https://dev.to/bytebodger/default-props-in-react-typescript-2o5o